util/gdb-dmtcp-utils.py

```
@@ -93,18 +95,17 @@ def is_recent_gdb():
   return is_recent_gdb.value
 # This should be used only for executable binaries.
 def file_base_address(file):
-  output = subprocess.run("readelf -S " + file + " | grep '\.text'",
-                          shell=True, stdout=subprocess.PIPE,
+  output = subprocess.run(["readelf", "-S", file], stdout=subprocess.PIPE,
```

Ruff (0.15.20)

[error] `subprocess` call: check for execution of untrusted input

(S603)